src/main/scala/com/fulcrumgenomics/bam/ErrorRateByReadPosition.scala

```diff
@@ -110,18 +111,18 @@ class ErrorRateByReadPosition
       logger.warning("No metrics generated (is your BAM empty?). Plots will not be generated.")
     }
     else {
-      val name = PathUtil.basename(input, trimExt=true).toString
-      Rscript.execIfAvailable(ScriptPath, metricsPath.toString, plotPath.toString, name) match {
+      val description = plotDescription(in, input)
+      Rscript.execIfAvailable(ScriptPath, metricsPath.toString, plotPath.toString, description) match {
         case Failure(e) => logger.warning(s"Generation of PDF plots failed: ${e.getMessage}")
         case _ => Unit
       }
     }
   }
 
   /** Computes the metrics from the BAM file. */
-  private[bam] def computeMetrics: Seq[ErrorRateByReadPositionMetric] = {
+  private[bam] def computeMetrics(source: Option[SamSource] = None): Seq[ErrorRateByReadPositionMetric] = {
     val progress = ProgressLogger(logger, verb="Processed", noun="loci", unit=50000)
-    val in       = SamSource(input, ref=Some(ref))
+    val in       = source.getOrElse(SamSource(input, ref=Some(ref)))
```

Comment:
This `source = None` argument confused me for a second.  Why not just have the argument be of type `SamSource` with a default value of `SamSource(input, ref=Some(ref))`?